atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/snapshot/KeyValueSnapshotReader.java

```diff
@@ -41,6 +42,7 @@
  * Although this interface performs user-level reads, internal writes may be performed (for example, as part of the
  * read protocol, to abort a long-running transaction).
  */
+@AutoDelegate
```

Comment:
I can't remember: what were we planning to do with this in the end?

Reply:
We're providing an implementation of KVSR that dual reads two delegate KVSR instances. Those delegates are lazily initialized, and the implementation relies on AutoDelegate https://pl.ntr/2nc

Reply:
Ah okay. Yeah I didn't realise the internal one used AutoDelegate. Thanks for the link